lib/operations/mongo_client_ops.js

```diff
@@ -117,7 +119,12 @@ function addListeners(mongoClient, topology) {
 
 function assignTopology(client, topology) {
   client.topology = topology;
-  topology.s.sessionPool = new ServerSessionPool(topology.s.coreTopology);
+
+  if (topology instanceof NativeTopology) {
+    topology.s.sessionPool = new ServerSessionPool(topology);
```

Comment:
because there was a sense that sessions should be a native specific feature, which at this point I think was misguided. Having said that, I'd prefer to punt moving this into `Topology`